core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -604,6 +596,14 @@ public long getStartTime() {
                             .stream()
                             .filter(e -> Regex.simpleMatch(pattern, e.getKey()))
                             .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+                } else {
```

Comment:
I'd probably keep the clauses in the order they were in before. Not a big deal though.